### DISCUSSION TITLE
How to package project as a jar or apk? i want get a apk witch can run with my phone...

### DISCUSSION THREAD

**Discussion by loveqianqian413** [AlmasB/FXGL#1291] [Q&A]
i try to package a jar with maven use config below:
`
<plugin>
<groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <version>3.10.1</version>
                <configuration>
                    <release>${source.version}</release>
                </configuration>
            </plugin>
            <plugin>
                <groupId>org.openjfx</groupId>
                <artifactId>javafx-maven-plugin</artifactId>
                <version>${jfx.maven.plugin.version}</version>
                <configuration>
                    <launcher>App</launcher>
                    <mainClass>${mainClassName}</mainClass>
                </configuration>
            </plugin>
`
and use java -jar  command, then it show "not found main class".
there 2 question:
1. how can i package a useable jar with maven? pls give me a config.
2. how can i package a apk? 
thanks a lot!
（my english is pool , execute me）

**Comment by chengenzhao** [comment]
这个问题没那么简单
不是说你想点个按钮，就自动给你打包好的
整个过程比较麻烦，一个简单的参考是fxgl games里面的geometry war
具体的build流程参考：https://github.com/AlmasB/FXGLGames#build-for-mobile
目前不支持win上的交叉编译，你要编译安卓上的apk，需要一台linux
不过可以通过github action来解决，但是github action的配置也很麻烦，这个项目可能会有帮助：
https://github.com/gluonhq/hello-gluon-ci

**Comment by sheeehy** [comment]
To package a usable JAR with Maven:
```
<build>
    <plugins>
        <plugin>
            <groupId>org.apache.maven.plugins</groupId>
            <artifactId>maven-jar-plugin</artifactId>
            <version>3.2.0</version>
            <configuration>
                <archive>
                    <manifest>
                        <addClasspath>true</addClasspath>
                        <mainClass>${mainClassName}</mainClass>
                    </manifest>
                </archive>
            </configuration>
        </plugin>
    </plugins>
</build>
```
To package an APK, you typically use Android Studio and the Android Gradle Plugin. However, if you're trying to run a JavaFX application on Android, there's a project called [Gluon](https://gluonhq.com/) which allows you to package and deploy JavaFX applications on mobile platforms,
